tests/test_streaming_download_manager.py

```diff
@@ -29,36 +37,33 @@
     ],
 )
 def test_xjoin(input_path, paths_to_join, expected_path):
-    from datasets.utils.streaming_download_manager import xjoin
-
     output_path = xjoin(input_path, *paths_to_join)
     assert output_path == expected_path
+    output_path = xpathjoin(Path(input_path), *paths_to_join)
+    assert output_path == Path(expected_path)
 
 
 def test_xopen_local(text_path):
-
     with xopen(text_path, encoding="utf-8") as f, open(text_path, encoding="utf-8") as expected_file:
         assert list(f) == list(expected_file)
+    with xpathopen(Path(text_path), encoding="utf-8") as f, open(text_path, encoding="utf-8") as expected_file:
```

Comment:
Since `Path` converts `://` to `:/` it would be nice to have a test with `zip://` and a local path, and also with a remote path

Reply:
This is indirectly tested by `test_as_posix`, do you agree?

Reply:
yup thanks !